x/tokenomics/keeper/update_relay_mining_difficulty.go

```diff
@@ -72,17 +76,29 @@ func (k Keeper) UpdateRelayMiningDifficulty(
 		}
 		k.SetRelayMiningDifficulty(ctx, newDifficulty)
 
-		// Output the appropriate log message based on whether the difficulty was
-		// initialized, updated or unchanged.
+		// Emit an event for the updated relay mining difficulty regardless of
+		// whether the difficulty changed or not.
+		relayMiningDifficultyUpdateEvent := types.EventRelayMiningDifficultyUpdated{
+			ServiceId:        serviceId,
+			PrevTargetHash:   prevDifficulty.TargetHash,
+			NewTargetHash:    newDifficulty.TargetHash,
+			PrevNumRelaysEma: prevDifficulty.NumRelaysEma,
+			NewNumRelaysEma:  newDifficulty.NumRelaysEma,
+		}
+		if err := sdkCtx.EventManager().EmitTypedEvent(&relayMiningDifficultyUpdateEvent); err != nil {
+			return err
+		}
+
+		// Output the appropriate log message based on whether the difficulty was initialized, updated or unchanged.
+		var logMessage string
 		if !found {
-			logger.Info(fmt.Sprintf("Initialized RelayMiningDifficulty for service %s at height %d with difficulty %x", serviceId, sdkCtx.BlockHeight(), newDifficulty.TargetHash))
-			continue
+			logMessage = fmt.Sprintf("Initialized RelayMiningDifficulty for service %s at height %d with difficulty %x", serviceId, sdkCtx.BlockHeight(), newDifficulty.TargetHash)
 		} else if !bytes.Equal(prevDifficulty.TargetHash, newDifficulty.TargetHash) {
-			// TODO_BLOCKER(@Olshansk, #542): Emit an event for the updated difficulty.
-			logger.Info(fmt.Sprintf("Updated RelayMiningDifficulty for service %s at height %d from %x to %x", serviceId, sdkCtx.BlockHeight(), prevDifficulty.TargetHash, newDifficulty.TargetHash))
+			logMessage = fmt.Sprintf("Updated RelayMiningDifficulty for service %s at height %d from %x to %x", serviceId, sdkCtx.BlockHeight(), prevDifficulty.TargetHash, newDifficulty.TargetHash)
 		} else {
```

Comment:
Nit: this might read slightly better as a `switch` statement.

Reply:
The condition for the first and second branch are kind of different though. Don't see how a switch fits.

Reply:
```
switch {
  case !found:
    // ...
  case !bytes.Equal(...):
    // ...
  default:
    // ...
}
```